### PR TITLE
Add saving correlation to EZ sweep

### PR DESCRIPTION
**Pull request by gwaygenomics**
This pull request is dependent on greenelab/tybalt#136 (the hash in ccc2ba1 is going to update). I will also likely rebase this PR - I will remove the WIP tag once this is done.

I add saving correlation (Pearson and Spearman) to both samples and genes before and after compression across different algorithms. I also update the README and analysis scripts accordingly.

### CODE REVIEW

**Comment by gwaygenomics** [comment]
Successful rebase - converted a rough-to-review PR into one that's hopefully more manageable!

File: 2.ensemble-z-analysis/scripts/train_models_given_z.py [hunk 1]
```diff
@@ -55,6 +59,65 @@
...
+def get_recon_correlation(df, recon_mat_dict, algorithm, cor_type, axis):
+    """
+    Get gene or sample correlations between input and reconstructed input
+
+    Arguments:
+    df - the input dataframe
+    recon_mat_dict - dictionary of different algorithms reconstructions
+    algorithm - string representing the compression algorithm
+    cor_type - string representing Pearson or Spearman correlations
+    axis - string representing genes or samples to calculate correlations for
```
Comment (danich1) [inline 2.ensemble-z-analysis/scripts/train_models_given_z.py]: Is there going to be future options for the axis parameter? if not wouldn't it be easier to just make a boolean?

File: 2.ensemble-z-analysis/README.md [hunk 2]
```diff
@@ -22,14 +23,30 @@ We compress gene expression data with the following algorithms:
...
+For each of the populations, we will also track performance of training and testing sets independently.
 
 1. Reconstruction Cost - Measures the binary cross entropy of input data to reconstruction
 2. Training History - For neural network models (ADAGE, Tybalt), save the training progress of each model
    * For Tybalt, the KL Divergence and Reconstruction Loss are saved separately
 3. Correlation of input sample to reconstructed sample - Measure how well certain samples traverse through the bottleneck.
    * Calculate Pearson and Spearman correlations
    * May reveal certain biases in sample reconstruction efficiency across algorithms
+4. Correlation of input gene profile to reconstructed gene profile - Measure how well certain genes are reconstructed through the bottleneck.
+   * Calculate Pearson and Spearman correlations
```
Comment (danich1) [inline 2.ensemble-z-analysis/README.md]: Why are you calculating the spearman correlation?

Comment (gwaygenomics) [reply 2.ensemble-z-analysis/README.md]: Non-normally distributed data should be modeled better with Spearman than Pearson. I can imagine a scenario in which we compare the differential (Spearman - Pearson) across algorithms and observe that the nonlinear neural networks are better when the data is "more" non-normal. Some similar observations were shown in [Cleary et al.](https://doi.org/10.1101/091926)

File: 2.ensemble-z-analysis/scripts/train_models_given_z.py (hunk 1 above)
Comment (gwaygenomics) [reply 2.ensemble-z-analysis/scripts/train_models_given_z.py]: yeah, that is a good point. I will update!

**Comment by gwaygenomics** [comment]
@danich1 - any follow-up comments?

**Pull request merged**